packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -131,6 +131,10 @@ export class Span implements api.Span, ReadableSpan {
     startTime?: api.TimeInput
   ): this {
     if (this._isSpanEnded()) return this;
+    if (this._spanLimits.eventCountLimit! === 0) {
+      api.diag.warn('No events allowed.');
+      return this;
+    }
     if (this.events.length >= this._spanLimits.eventCountLimit!) {
```

Comment:
I think it might be easier to change this if statement instead of adding another
```suggestion
    if (this._spanLimits.eventCountLimit && this._spanLimits.eventCountLimit < this.events.length) {
```

Reply:
That if statement won't work:
1. If the limit is set to 0 no event will be dropped (due to first part of the if condition) and one will be added every time. So, if you try adding 10 events you'll end up with 10 events
2. If the limit is set to a value higher than 0 (let's say 10), and you try to add 11th event it will be added (due to the second part of the if statement) and still no event will be dropped.

The point is to return from the function immediately and not trying to add/drop an event when the limit is 0.

Reply:
ah i misread what was going on